__tests__/app.test.js

```diff
@@ -36,17 +44,82 @@ describe("GET/api/articles/:article_id", () => {
     return request(app)
       .get("/api/articles/1")
       .expect(200)
-      .then(({ body }) => {
-        expect.objectContaining(body);
+      .then(({ body: article }) => {
+        expect(article).toHaveLength(1);
+        article.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              article_id: expect.any(Number),
+              title: expect.any(String),
+              topic: expect.any(String),
+              author: expect.any(String),
+              body: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number)
+            })
+          );
+        });
       });
   });
-  test("404:article id not in database", () => {
+  test("400:article id not in database", () => {
     return request(app)
       .get("/api/articles/1123")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Id not found" );
+      });
+  });
+});
+
+describe("GET/api/users/", () => {
+  test("200: return users", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body: users }) => {
+        expect(users).toHaveLength(4);
+        users.forEach((users) => {
+          expect(users).toEqual(
+            expect.objectContaining({
+              username: expect.any(String),
+              name: expect.any(String),
+              avatar_url: expect.any(String)
+            })
+          );
+        });
+      });
+  });
+  test("404: mispelt endpoint", () => {
+    return request(app)
+      .get("/api/user")
       .expect(404)
-        .then(({ body }) => {
-          console.log(body)
-        expect.objectContaining(body);
+      .then(({ misspeltUsers }) => {
+        expect(misspeltUsers).toEqual(undefined);
       });
   });
 });
+
+// describe.only("GET/api/articles/:article_id", () => {
```

Comment:
I'm not sure what ticket this PR covers? You seem to have skipped patchArticlesById and gone straight to getUsers. That's not a problem in this case, although you shouldn't push any code that isn't in use / you don't want to be merged into main. Therefore, please remove this commented-out block before merging.

Reply:
If this is to cover patchArticleById then you _really_ need to reinstate this test and add some error handling.